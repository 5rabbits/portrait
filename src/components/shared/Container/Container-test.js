import { React, expect, mount } from 'test/helper'
import Container from 'shared/Container'

describe('Container', () => {
  it('should render with minimum props', () => {
    const component = mount(<Container />)

    expect(component).to.be.ok
  })
})
