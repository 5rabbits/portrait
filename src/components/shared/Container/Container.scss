@import '~styles/core';

.Container--normal,
.Container--large,
.Container--fluid {
  @include clearfix;
  margin-right: auto;
  margin-left: auto;
  padding-left: $gutter;
  padding-right: $gutter;
}

.Container--normal {
  max-width: $container-normal-width;
}

.Container--large {
  max-width: $container-large-width;
}

.Container--fluid {
  max-width: 100%;
}
