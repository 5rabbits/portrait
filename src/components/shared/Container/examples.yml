code: >
  <Container style={{ background: '#fff' }}>
    Some content
  </Container>
