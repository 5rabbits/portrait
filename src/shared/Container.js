import React from 'react'

const Container = props =>
  <div className="container-page">
    {props.children}
  </div>

export default Container
