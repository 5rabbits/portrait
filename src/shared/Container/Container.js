import React from 'react'
import childrenPropType from 'propTypes/children'

const Container = props =>
  <div className="container-page">
    {props.children}
  </div>

Container.propTypes = {
  children: childrenPropType,
}

export default Container
