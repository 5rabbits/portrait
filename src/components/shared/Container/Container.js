import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import './Container.scss'

/**
 * Wrapper component for the main page contents. It includes some horizontal padding to
 * avoid the content for reaching the screen edges.
 *
 * This component will have some responsive properties in the near future üòÅ.
 */
export default class Container extends PureComponent {
  static propTypes = {
    /**
     * The contents to be displayed.
     */
    children: PropTypes.node.isRequired,

    /**
     * CSS class assigned to the container.
     */
    className: PropTypes.string,

    /**
     * The size of the container. Use `fluid` for content that should use the full screen width.
     */
    size: PropTypes.oneOf([
      'fluid', 'large', 'normal',
    ]).isRequired,
  }

  static defaultProps = {
    className: null,
    size: 'normal',
  }

  render() {
    const { children, className, size, ...other } = this.props

    return (
      <div
        {...other}
        className={cx(`Container--${size}`, className)}
        >
        {children}
      </div>
    )
  }
}
