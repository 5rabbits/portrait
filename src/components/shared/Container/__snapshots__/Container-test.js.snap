// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render with minimum props 1`] = `
<Container
  className={null}
  size="normal"
>
  <div
    className="Container Container--normal"
  >
    Test
  </div>
</Container>
`;
